python/CHANGELOG.rst

```diff
@@ -12,6 +12,11 @@ In development
 
 **New features**
 
+- The Kendall-Colijn tree distance metric now operates on samples instead of
```

Comment:
Is there any potential breakage here? I think we were assuming before that leaves == samples, so it's probably the same thing ultimately, except we didn't support internal samples before.

Reply:
> Is there any potential breakage here?

No I don't believe so.



> I think we were assuming before that leaves == samples, so it's probably the same thing ultimately, except we didn't support internal samples before.

This is pretty much right. We were enforcing that samples were a subset of leaves, and breaking on leaves that were not samples. So any previously working code must have had samples == leaves. There's "anti-breakage" in that code with non-sample leaves that used to break will now silently pass. The difference between leaves and samples are in the docs, I'm unsure whether it should be reiterated here.

Reply:
We weren't actually enforcing non-sample leaves before through, were we? I.e., I don't think it would have thrown an error if there were leaves that were not samples (as opposed to samples that were not leaves). So, we didn't actually change very much, I think (need to be quite precise about what has changed).

Reply:
You're right we were not. If we're just sticking to enforcement, we could just say "now permits internal samples".

Reply:
I think that's the right approach - rather than saying we're doing something different now, say we're generalising what was there before.